File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int n=nums.length,max=0;
        int dp[]=new int[n];
        dp[0]=1;
        for(int i=1;i<n;++i)
        {
            dp[i]=1;
            for(int j=0;j<i;++j)
            {
                if(nums[i]>nums[j]&&dp[i]<dp[j]+1)
                    dp[i]=dp[j]+1;
            }
            
        }
        for(int i=0;i<dp.length;++i)
            if(dp[i]>max)
                max=dp[i];
        return max;
    }
}

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp
class Solution {
public:
    
//     int helper(int i,int prev,vector<int>&arr, int n)
//     {
//         if(i==n) return 0;
//         int len=0+helper(i+1,prev,arr,n);
//         if(arr[i]>arr[prev]||prev==-1)
//              len=max(len,1+helper(i+1,i,arr,n));
            
//      return len;
//     }
    int lengthOfLIS(vector<int>& arr) {
         int n=arr.size();
        int lis[n]; 

	lis[0] = 1; 

	for (int i = 1; i < n; i++ ) 
	{ 
		lis[i] = 1; 
		for (int j = 0; j < i; j++ ) 
			if ( arr[i] > arr[j] && lis[i] < lis[j] + 1) 
				lis[i] = lis[j] + 1; 
	} 

	return *max_element(lis, lis+n); 
        
        //return helper(0,-1,arr,n); 
    }
};